### PR TITLE
add back the --dry-run=false flag for sinker and horologium

### PR DESCRIPTION
**Pull request by krzyzacy**
after https://github.com/kubernetes/test-infra/pull/11267
/hold

/assign @cjwagner

### CODE REVIEW

**Comment by cjwagner** [comment]
@krzyzacy What is the context for this change? Are these components currently running in dry-run mode or is the default for the flag changing?

**Comment by krzyzacy** [comment]
@cjwagner this is basically https://github.com/kubernetes/test-infra/pull/11092, which is needed after we migrate sinker and horologium to the new k8s client. The flag is added from https://github.com/kubernetes/test-infra/pull/11066 and https://github.com/kubernetes/test-infra/pull/11086

File: prow/cluster/horologium_deployment.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ spec:
         image: gcr.io/k8s-prow/horologium:v20190125-2aca69d
         args:
         - --job-config-path=/etc/job-config
+        - --dry-run=false
```
Comment (fejta) [inline prow/cluster/horologium_deployment.yaml]: Does this flag exist in `gcr.io/k8s-prow/horologium:v20190125-2aca69d`?

We need to make this flag default to true for some time period before we require people to set it.

Comment (krzyzacy) [reply prow/cluster/horologium_deployment.yaml]: you mean make the flag default to false for some period?

**Pull request closed without merging**